docroot/sites/admissions.uiowa.edu/modules/admissions_core/admissions_core.module

```diff
@@ -696,6 +704,36 @@ function admissions_core_preprocess_node(&$variables) {
             }
             break;
         }
+        break;
+
+      case 'transfer_tips':
+        // Check if we have any major (2 plus 2) content pieces
+        // that reference the transfer tip's area of study.
+        $area_of_study_id = $node->field_transfer_tips_aos->target_id;
+        $query = \Drupal::entityQuery('node')
+          ->condition('type', 'major')
+          ->condition('status', 1)
+          ->condition('field_major_area_of_study', $area_of_study_id, '=');
+        // We only really need to know if there areas of study,
+        // and not which or how many, because the link will just be
+        // based on the aos id.
+        $count = $query->count()->execute();
+        if ($count > 0) {
+          $area_of_study_url = \Drupal::service('path_alias.manager')->getAliasByPath('/node/' . $area_of_study_id);
+          // Get the link to the 2 Plus 2 community college aggregator page
+          // using the last segment of the aos url, which is constructed
+          // from the node title.
+          $url = \Drupal::service('path_alias.manager')->getAliasByPath('/2plus2/majors/' . basename($area_of_study_url));
```

Comment:
This feels like a "works 95% of the time" type solution here. There might be some cases with titles like "Asian Languages and Literature" which get handled slightly differently in how the path alias gets constructed and the contextual filter query parameter gets taken in...Need to check that.

An alternative would be to grab the node title, lower-case it, and replace spaces with hyphens (following the method that the contextual filter should be using).

Reply:
Update: it does break on those examples.

Reply:
Would it be too bold to assume we could use pathauto's alias cleaner to get the same path? We are using `pathauto.alias_cleaner` in several configuration forms (SiteNow Articles, People, etc.)

Reply:
oh maybe i read this wrong. it isn't just matching the aos path component but massaging it for contextual filtering...

Reply:
Yep...the contextual filtering is based on the title and not the path, which also means that the 2 plus 2 view page has a path a la "2plus2/majors/asian-languages-and-literature" where the aos has "academics/asian-languages-literature". Which is an annoying discrepancy, as well as potentially problematic in spots.

Reply:
I'm wondering if it's possible to contextually filter on path alias rather than title..?

Reply:
docroot/core/modules/views/src/Plugin/views/argument/StringArgument.php